### ISSUE TITLE
Cannot Connect

### ISSUE THREAD

**Issue by znbaboy** [shrimqy/Seki#4]
Installed latest version on Windows + Android and trying to connect does nothing. See video

https://github.com/user-attachments/assets/08cfe043-8ea5-4595-9df7-a28b4f2a0c03

**Comment by shrimqy** [comment]
Are you perhaps using on a virtual machine? Plus it would be good if you can provide some logs.

**Comment by shawny43** [comment]
How to provide logs when you are on Android or Windows. Another thing, why when i tried to connect i receive nothing on my desktop. 

Unfortunately im using a Windows 11 custom pc and Pixel 6a.

**Comment by shrimqy** [comment]
You can collect the logs by running `adb logcat > log.txt` on the terminal and ctrl + c to stop collecting. You need to enable usb debugging as well.
Windows 11 and pixel 6a shouldn't be an issue here as well. If I can somehow reproduce the issue I will fix as soon as I can.

**Comment by shawny43** [comment]
[logcat.txt](https://github.com/user-attachments/files/17285968/logcat.txt)

Here the log, it's seem too long and take space to be posted there. Even pastbin are pretty limited. 

Like this host of the topic, i have the same issue that my pc doesn't seem detect my device request.

**Comment by shrimqy** [comment]
It does seem to connect to a local ip address as it supposed to, can you check if there are any firewalls present and it is not blocking the windows app? Multiple network interfaces can also be a problem here.

**Comment by shawny43** [comment]
> It does seem to connect to a local ip address as it supposed to, can you check if there are any firewalls present and it is not blocking the windows app? Multiple network interfaces can also be a problem here.

Unfortunately, it's seem my firewall on my PC have already authorized the desktop app. Also, Android phones doesn't have firewall settings. 

![image](https://github.com/user-attachments/assets/4d0e6df2-449f-4dd7-a98c-0e0770adf930)

Also, my operator reset the public ip of my network on each log-in so i can't help at the moment. It's seem you spoke about static IP wich is not the case.

**Comment by shrimqy** [comment]
Your dynamic public ip wouldn't also be an issue here because the app basically uses the local ip given by the router to make the connection. Or maybe it is the local ip that is dynamic which might pose a problem here but I am unable to actually verify it. 
Another possibility is your router has enabled AP isolation?

**Comment by shawny43** [comment]
Not the case, we rent a router and doesn't block local IP normally. KDE Connect, Windows Link doesn't have this issue. Seem difficult to see where is the source of the problem. My modem/router are an Hiltron from my local Operator. Also chromecast and other devices work locally so it's not supposed to be that AP issue.

**Comment by Deltastisch** [comment]
I also have problems connecting. I am using gnirehtet to connect my phone to the internet over USB. When I do this I can see my phone on the Windows app but do not see my Laptop in the Android App.

**Comment by shrimqy** [comment]
>If I try to connect over Wi-Fi, everything works.

I see, since you can connect over WI-FI, after pairing go into the device details page, add a custom Ip and select it as your preferred IP. Let me know if that fixes it.

**Comment by znbaboy** [comment]
Just tried on latest version,

I get infinite loading when trying to connect my device. I am able to see it on the Windows app with the correct PIN, but it just doesn't connect. No VPN or anything

**Comment by shrimqy** [comment]
>It works but only connects through the Settings GUI on the Windows app. Trying to connect via the "setup wizard" caused it to never establish a connection despite seeings the device in the app on my network.

Could you produce some logs if you can, It would be really helpful.

>After establishing connection successfully, the Windows app crashes. Cannot connect my phone without the app crashing on PC.

This seems to be an entirely different issue.. Umm I do have assumptions, could you not enable the Remote Storage access on android just disable it and try again? 

In the about page (Windows), there should be a "Open logs" option, and could you upload the logs here? (Ofcourse check for any sensitive information before uploading)

**Comment by znbaboy** [comment]
> Could you produce some logs if you can, It would be really helpful.

[Log20250205.log](https://github.com/user-attachments/files/18677173/Log20250205.log)

The logs dont seem to record the crashes, but I was having a hard time getting the device to connect again so maybe those logs will help

> This seems to be an entirely different issue.. Umm I do have assumptions, could you not enable the Remote Storage access on android just disable it and try again?

Tried disabling "All files access" for the app on Android, crash still persisted

**Comment by shrimqy** [comment]
>[Log20250205.log](https://github.com/user-attachments/files/18677173/Log20250205.log)

These logs don't provide any actual info about the crash like you said, are you sure this was the only log file in the directory?

>UPDATE: ...now it seems to have connected without crashing with storage access enabled.... 

Well that is good news. Could you share those latest logs? 

>there doesn't seem to be any rhyme or reason to this madness

I thought I had found all the plausible crashing scenarios but I was dead wrong ig.

**Comment by znbaboy** [comment]
> These logs don't provide any actual info about the crash like you said, are you sure this was the only log file in the directory?

![Image](https://github.com/user-attachments/assets/3c978335-1e4a-4a4d-a297-edde04cc84ad)

Yes its the only log there.

> Well that is good news. Could you share those latest logs?

The latest logs dont show anything new. Here's the one new line: 

```2025-02-05 13:56:25.007 -05:00 [ERR] Unhandled exception occurred
System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request..)
 ---> System.IO.IOException: Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request..
 ---> System.Net.Sockets.SocketException (995): The I/O operation has been aborted because of either a thread exit or an application request.
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
   at System.Net.Security.SslStream.EnsureFullTlsFrameAsync[TIOAdapter](CancellationToken cancellationToken, Int32 estimatedSize)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](Memory`1 buffer, CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask`1.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
   --- End of inner exception stack trace ---
```

**Comment by shrimqy** [comment]
>Yes its the only log there.

Weird but I will look into this thank you.

>Also I had to re-pair my phone to my PC because I deleted the app. The paring process takes extremely long for some reason but does work.

Yes if you were to auto-connect, it does take a long time, I do need to find a new way or reduce the timeout, but if you know which IP works then the manual connection would connect instantly.

**Comment by shawny43** [comment]
It's me or the same issue is back again. I saw the same issue came back again? Did something goes wrong? I made the same ticket in the past and now im not alone with this case.

**Comment by Deltastisch** [comment]
> > If I try to connect over Wi-Fi, everything works.
> 
> I see, since you can connect over WI-FI, after pairing go into the device details page, add a custom Ip and select it as your preferred IP. Let me know if that fixes it.

I tried to do a custom ip on my laptop, but I didn't have internet.
Maybe the reason I can't connect is because gnirehtet uses a VPN connection on my phone to share internet

**Comment by shrimqy** [comment]
>I tried to do a custom ip on my laptop, but I didn't have internet.
Maybe the reason I can't connect is because gnirehtet uses a VPN connection on my phone to share internet

So umm, this is such a peculiar use case, how does that work though? I'm not too much familiar with gnirehtet but as far as I can see it seems to be using ipv4 traffic as well? I don't see a reason why it doesn't work tbh, anyways I will try to use and reproduce it.

**Comment by shrimqy** [comment]
>For me, the app goes correctly through the setup & connection egotiation, but then can't connect the main process.

@HighPriest sorry for the delay, but this issue is not related to what you have, refer #10 if that fixes your issue if not create a new one with the logs.

**Comment by znbaboy** [comment]
using Windows app 1.0.4 and Android app 1.0.7 works perfectly

auto connect worked seamlessly as well. I can finally daily this app! awesome

**Comment by hiiamboris** [comment]
I can't connect either. On Windows I see the Android device, but there's no "connect" button or anything. On Android at the same time it shows "No devices found", so there's definitely a bug in the Android version. Trying to add the Windows IP to the Android app shows it as unreachable. Both devices share the same network (in this case they're connected to the same Wireguard server). I added Sefirah to Defender's whitelist into both Incoming and Outgoing rules for all networks.

**Comment by shrimqy** [comment]
@hiiamboris try checking if the UDP broadcasts are being transmitted. You can use Wireshark or something similar to monitor the traffic, the port is 5149. Try the same on Android using some UDP listener. I almost thought I had fixed all the connection issues haha-. Let me know what you find.

**Comment by hiiamboris** [comment]
I'm not ready to bother with Wireshark :/ but **NekoRay on Windows doesn't show any UDP broadcasts**. From the Android side I see passed UDP broadcasts using Rethink. Maybe that's why I see the Android device on the PC side.

I've tried VPNs on an off, Defender on and off, same results. Under some settings I'm able to see something like **"Ping 2ms" in custom devices** list on Android side, but that doesn't help me connect. Windows side still doesn't see anything.

**Comment by shrimqy** [comment]
@hiiamboris All right, I think I found the issue with why the Android is not discovering the Windows device, I had apparently commented out a piece of code, which was the backup discovery method. I suppose no one found an issue until now. No matter, I'm kinda surprised how the UDP isn't working as intended, but well if this works it doesn't really matter ig.

[sefirah-android-discovery-fix.zip](https://github.com/user-attachments/files/19552825/sefirah-discovery-fix.zip)

**Comment by hiiamboris** [comment]
Worked @shrimqy ☺
Very cool 👏 The only sane clipboard sharing utility I can find around, that's for sure.

Another bug I'm seeing is if I disconnect on Android I can't reconnect back until I kill and restart the app.

Also would be nice to be able to manually rename the devices instead of having just PC username and Android model.

**Comment by shrimqy** [comment]
>Worked

All right, I'm SO glad it worked.

>Another bug I'm seeing is if I disconnect on Android I can't reconnect back until I kill and restart the app.

I will try to investigate this, thanks. It would be really great if you could open an issue with the logs.


>Also would be nice to be able to manually rename the devices instead of having just PC username and Android model.

I will add that for the PC later, I just happened to forget about that (again). For Android, you can rename the device in the settings tab.

**Comment by hiiamboris** [comment]
Another issue is when it loses connection (e.g. I restart the VPS where Wireguard is set up), I keep seeing Android showing that it's connected, PC showing it's disconnected, and of course no communication is made. I had to restart both PC and Android apps to fix that. Preferably both apps should try to restore the connection on their own.

**Comment by shrimqy** [comment]
>I keep seeing Android showing that it's connected, PC showing it's disconnected, and of course no communication is made.

The general way of detecting a TCP socket failing is tedious. Implementing a different mechanism would not be worth the use of resources tbh so I just left it as it is. Though I will see if there is some other way of doing it. For now, it will work if the peer disconnects correctly. Most often, just restarting the connection fixes it, and there is no need to restart the apps. The auto-reconnection in your case would not work, since the udp broadcasts are not reaching your Android for some reason. 

>Where do I find the logs? There's no link in the Android app as far as I can tell.

I haven't implemented getting the logs yet, you would need to use adb for that. You don't need to if it is too much of a hassle.

**Comment by hiiamboris** [comment]
I'll just open an issue then without the logs :)
